frappe/public/js/form_builder/store.js

```diff
@@ -0,0 +1,38 @@
+import { defineStore } from "pinia";
+import { create_layout } from "./utils";
+import { nextTick } from "vue";
+
+export const useStore = defineStore("store", {
+	state: () => ({
+		doctype: "",
+		fields: [],
+		docfields: [],
+		layout: {},
+		dirty: false,
+	}),
+	actions: {
+		is_dirty() {
+			return this.dirty;
+		},
+		fetch() {
```

Comment:
This could just become an `async` function, right? That may help cleanup the callbacks on `frappe.model.with_doctype` and `nextTick`